projects/plugins/jetpack/class.jetpack-user-agent.php

```diff
@@ -5,660 +5,7 @@
-	 * - Facebook 4020.0 (iPhone; iPhone OS 5.0.1; fr_FR)
-	 * - Mozilla/5.0 (iPhone; U; CPU iPhone OS 5_0 like Mac OS X; en_US) AppleWebKit (KHTML, like Gecko) Mobile [FBAN/FBForIPhone;FBAV/4.0.2;FBBV/4020.0;FBDV/iPhone3,1;FBMD/iPhone;FBSN/iPhone OS;FBSV/5.0;FBSS/2; FBCR/O2;FBID/phone;FBLC/en_US;FBSF/2.0]
-	 * - Mozilla/5.0 (iPhone; CPU iPhone OS 5_1_1 like Mac OS X) AppleWebKit/534.46 (KHTML, like Gecko) Mobile/9B206 [FBAN/FBIOS;FBAV/5.0;FBBV/47423;FBDV/iPhone3,1;FBMD/iPhone;FBSN/iPhone OS;FBSV/5.1.1;FBSS/2; FBCR/3ITA;FBID/phone;FBLC/en_US]
-	 *
-	 * @deprecated 8.7.0 Use Automattic\Jetpack\Device_Detection\User_Agent_Info
-	 */
-	public static function is_facebook_for_iphone() {
-		_deprecated_function( __METHOD__, 'Jetpack 8.7', '\Automattic\Jetpack\Device_Detection\User_Agent_Info::is_facebook_for_iphone from the `automattic/jetpack-device-detection` package' );
-		return User_Agent_Info::is_facebook_for_iphone();
-	}
-
-	/**
-	 * Detects if the current UA is Facebook for iPad
-	 * - Facebook 4020.0 (iPad; iPhone OS 5.0.1; en_US)
-	 * - Mozilla/5.0 (iPad; U; CPU iPhone OS 5_0 like Mac OS X; en_US) AppleWebKit (KHTML, like Gecko) Mobile [FBAN/FBForIPhone;FBAV/4.0.2;FBBV/4020.0;FBDV/iPad2,1;FBMD/iPad;FBSN/iPhone OS;FBSV/5.0;FBSS/1; FBCR/;FBID/tablet;FBLC/en_US;FBSF/1.0]
-	 * - Mozilla/5.0 (iPad; CPU OS 6_0 like Mac OS X) AppleWebKit/536.26 (KHTML, like Gecko) Mobile/10A403 [FBAN/FBIOS;FBAV/5.0;FBBV/47423;FBDV/iPad2,1;FBMD/iPad;FBSN/iPhone OS;FBSV/6.0;FBSS/1; FBCR/;FBID/tablet;FBLC/en_US]
-	 *
-	 * @deprecated 8.7.0 Use Automattic\Jetpack\Device_Detection\User_Agent_Info
-	 */
-	public static function is_facebook_for_ipad() {
-		_deprecated_function( __METHOD__, 'Jetpack 8.7', '\Automattic\Jetpack\Device_Detection\User_Agent_Info::is_facebook_for_ipad from the `automattic/jetpack-device-detection` package' );
-		return User_Agent_Info::is_facebook_for_ipad();
-	}
-
-	/**
-	 *  Detects if the current UA is WordPress for iOS.
-	 *
-	 * @deprecated 8.7.0 Use Automattic\Jetpack\Device_Detection\User_Agent_Info
-	 */
-	public static function is_wordpress_for_ios() {
-		_deprecated_function( __METHOD__, 'Jetpack 8.7', '\Automattic\Jetpack\Device_Detection\User_Agent_Info::is_wordpress_for_ios from the `automattic/jetpack-device-detection` package' );
-		return User_Agent_Info::is_wordpress_for_ios();
-	}
-
-	/**
-	 * Detects if the current device is an iPad.
-	 * They type can check for any iPad, an iPad using Safari, or an iPad using something other than Safari.
-	 *
-	 * Note: If you want to check for Opera mini, Opera mobile or Firefox mobile (or any 3rd party iPad browser),
-	 * you should put the check condition before the check for 'iphone-any' or 'iphone-not-safari'.
-	 * Otherwise those browsers will be 'catched' by the ipad string.
-	 *
-	 * @param string $type iPad type.
-	 *
-	 * @deprecated 8.7.0 Use Automattic\Jetpack\Device_Detection\User_Agent_Info
-	 */
-	public static function is_ipad( $type = 'ipad-any' ) {
```

Comment:
This function is used by the following w.org plugins:

- [ ] `infinite-transporter`: themes/twentyfourteen.php,30
- [ ] `jp-custom-css`: custom-css/custom-css.php,31
- [ ] `themedy-toolbox`: custom-css/custom-css/custom-css.php,26
- [ ] `wpvkp-ultimate-shortcodes`: custom-css/custom-css/custom-css.php,26

I have a feeling a couple of these are because places have incorporated our code into theirs, but should check it out.

Reply:
- `infinite-transporter` Gated, but in a way that doesn't help (`jetpack_is_mobile` is the gate). Last update 7 yrs ago, 90+ installs which seem to include ~20 Jetpack sites.
- `jp-custom-css` No longer maintained. Like the above, gated by `jetpack_is_mobile`. No longer available (closed plugin). Installed on 25 Jetpack sites.
- `themedy-toolbox`: Last update 3 years ago, 1000+ installs. Seems to be the same code as `jp-custom-css` with a `jetpack_is_mobile` gate. 140 Jetpack installs.
- `wpvkp-ultimate-shortcodes`: Closed plugin. 3 Jetpack installs. Same code as above, `jetpack_is_mobile` gate.